fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/BaseCompositeAvailability.java

```diff
@@ -16,30 +16,39 @@
  */
 public abstract class BaseCompositeAvailability implements Availability {
 
+    private final Set<Availability> sourceAvailabilities;
+    private final Set<TableName> dataSourcesNames;
+
+    /**
+     * Constructor.
+     *
+     * @param availabilityStream  A potentially ordered stream of availabilities which supply this composite view
+     */
+    public BaseCompositeAvailability(Stream<Availability> availabilityStream) {
+        sourceAvailabilities = StreamUtils.toUnmodifiableSet(availabilityStream);
+        dataSourcesNames = StreamUtils.toUnmodifiableSet(sourceAvailabilities.stream()
+                .map(Availability::getDataSourceNames).flatMap(Set::stream)
```

Comment:
This wrapping confused me. Can we move the `sourceAvailabilities.stream()` down a line? The way I read it initially, it looked like we were trying to assign a `Stream` to `dataSourceNames` and forgetting to `collect`.